### DISCUSSION TITLE
SSO login / user store

### DISCUSSION THREAD

**Discussion by torstenlemke** [directus/directus#9866] [Archive]
Hello guys & girls,

I've been struggling to set up the auth via github, though it seems to be working now :)

One question that isn't clear in the docs is the following:

where is the difference between AUTH_PROVIDERS and OAUTH_PROVIDERS??
I got it showing me the `login with github` when setting it to a config like

```
AUTH_PROVIDERS="github"

AUTH_GITHUB_DRIVER="oauth2"
AUTH_GITHUB_CLIENT_ID="aölsjdhgfhjklö"
AUTH_GITHUB_CLIENT_SECRET="aöskljhduiaguhsidjoakpüosdajdhvgbjknlmöä#a"
AUTH_GITHUB_AUTHORIZE_URL="https://github.com/login/oauth/authorize"
AUTH_GITHUB_ACCESS_URL="https://github.com/login/oauth/access_token"
AUTH_GITHUB_PROFILE_URL="https://api.github.com/user"
```

though I am a bit lost where my local provider went 😆 

Also how do I assign the external user (oauth) from github to the local one or how can I sign up a user with a special perm via oauth?

This is (at lest for me) just not clear and confuses me a lot :(

Thanks in advance 👍🏻 

Best Regards
Torsten

**Comment by rijkvanzanten** [comment]
OAUTH_PROVIDERS is what we had before the production release 🙂 

We replaced the oauth implementation with an "auth adapter" based system that allows for any auth flow to be integrated in a standardized way, hence the change from `OAUTH_PROVIDERS` to `AUTH_PROVIDERS` 👍🏻 

> Also how do I assign the external user (oauth) from github to the local one or how can I sign up a user with a special perm via oauth?

As of right now, a user can only login with the auth provider they signed up with. See https://github.com/directus/directus/issues/9331 for an ongoing issue on how to implement what you're asking for most efficiently
